File: matrix.java

// Time Complexity : O(r+c)
// Space Complexity : O(1)
// Did this code successfully run on Leetcode : Yes
class Solution {
        public boolean searchMatrix(int[][] matrix, int target) {
            int r = matrix.length;
            int c = matrix[0].length;
            int i=0, j=c-1;
           
            while(i<=r-1&&j>=0){
                int val = matrix[i][j];
                if(val == target)
                    return true;
                else if(matrix[i][j]>target){
                    j--;
                }
                else if(matrix[i][j]<target){
                    i++;
                }
               
            }
             return false;
        }
    }

File: matrix.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        m=len(matrix)
        n=len(matrix[0])
        
        r=0
        c=n-1
        while(r<m and c>=0):
            if(matrix[r][c]==target):
                return True
            elif(matrix[r][c]>target):
                c-=1
            else:
                r+=1
                
        return False
            
        